testpar/t_select_io_dset.c

```diff
@@ -3446,16 +3571,16 @@ test_no_selection_io_cause_mode(const char *filename, hid_t fapl, uint32_t test_
          * As the xfer mode is H5FD_MPIO_INDEPENDENT, this will call
          * H5FD__read/write_from_selection() triggering H5D_SEL_IO_NO_VECTOR_OR_SELECTION_IO_CB.
          */
-        no_selection_io_cause_read_expected |= H5D_SEL_IO_NO_VECTOR_OR_SELECTION_IO_CB;
+        //    no_selection_io_cause_read_expected |= H5D_SEL_IO_NO_VECTOR_OR_SELECTION_IO_CB;
 
         /* Exception case: This will turn off selection I/O landing at H5FD__mpio_write() */
         if ((test_mode & TEST_TCONV_BUF_TOO_SMALL) && !(test_mode & TEST_IN_PLACE_TCONV))
             no_selection_io_cause_write_expected |= H5D_SEL_IO_TCONV_BUF_TOO_SMALL;
         else
-            no_selection_io_cause_write_expected |= H5D_SEL_IO_NO_VECTOR_OR_SELECTION_IO_CB;
+            //       no_selection_io_cause_write_expected |= H5D_SEL_IO_NO_VECTOR_OR_SELECTION_IO_CB;
```

Comment:
Looks like commenting out this line will cause the setting of selection I/O below to only occur in the else clause. Is that intended?

Reply:
I reworked this section, I think it's cleaner now